### ISSUE TITLE
Get gyro rate via central finite difference on gyro heading instead

### ISSUE THREAD

**Issue by calcmogul** [wpilibsuite/sysid#328]
The gyro rate is way too noisy because it typically doesn't include the filtering the firmware generally does to get heading.
[sysid_data20220125-192839.json.txt](https://github.com/wpilibsuite/sysid/files/7938196/sysid_data20220125-192839.json.txt)
![Screenshot_20220125_164243](https://user-images.githubusercontent.com/2748555/151084479-79149e23-5e4c-4ab2-a71e-20b224bfacdc.png)

**Comment by calcmogul** [comment]
I tried doing a central finite difference, but the heading data has too many step changes, so the numerical differentiation has banding in it. I tried adding an acausal moving average to it, but that caused spikes at the step locations.

A single-pole IIR filter seemed to work OK, but now that I look at the plots above, it seems that the gyro angle is backwards from what sysid assumes, which is causing sysid to produce negative gains.
